python/modules/IcePy/ValueFactoryManager.h

```diff
@@ -16,51 +16,46 @@ namespace IcePy
 
     bool initValueFactoryManager(PyObject*);
 
-    class FactoryWrapper final : public Ice::ValueFactory
+    // The IcePy C++ value factory abstract base class.
+    struct ValueFactory
+    {
+        virtual std::shared_ptr<Ice::Value> create(std::string_view) = 0;
+    };
+    using ValueFactoryPtr = std::shared_ptr<ValueFactory>;
+
+    // Adapts a Python value factory to our C++ value factory.
+    class CustomValueFactory final : public ValueFactory
     {
     public:
-        FactoryWrapper(PyObject*);
-        ~FactoryWrapper();
+        CustomValueFactory(PyObject*);
+        ~CustomValueFactory();
 
         std::shared_ptr<Ice::Value> create(std::string_view) final;
 
         PyObject* getValueFactory() const;
-        void destroy();
 
-    protected:
+    private:
         PyObject* _valueFactory;
     };
 
-    using FactoryWrapperPtr = std::shared_ptr<FactoryWrapper>;
+    using CustomValueFactoryPtr = std::shared_ptr<CustomValueFactory>;
 
-    class DefaultValueFactory final : public Ice::ValueFactory
+    // The default Python value factory as a C++ value factory.
+    class DefaultValueFactory final : public ValueFactory
     {
     public:
         std::shared_ptr<Ice::Value> create(std::string_view) final;
-
-        void setDelegate(const Ice::ValueFactoryPtr&);
```

Comment:
The whole delegate thing did not make sense to me, so I removed it.

If the user registers a value factory with type ID "", we can just store it in the C++ map with this string, like any other type ID.